### DISCUSSION TITLE
Add support for Mega AVR chips

### DISCUSSION THREAD

**Discussion by anabolyc** [micronucleus/micronucleus#275] [Ideas]
Hi Team,

I had some great fun with the Attiny85 and micronucleos in the past, now slowly shifting towards newer 1-series and 2-series megaAVR Tinys. I was wondering, is it realistcally possible to port mironucleos to modern Tinys?

Why?
- They are x-times cheaper
- They have much more to give
- They are available. I know, old tinys are back in stock, but not in quantities, and still way too expensive by today's standards.

I did small investigation, and although it looks promising on the high level, I found that ubsdrv is quite heavily hardwired to old architecture (I stopped in front of wall of assembly code, directly manipulating ports, that's way beyond my reach). 

Newer Tinys are not exactly mistery, Spence Konde [released](https://github.com/SpenceKonde/megaTinyCore) core for the whole series. Perhaps someone more experinced here can tell, how hard can it be?

**Comment by cpldcpu** [comment]
To my knowledge there is nothing that would fundamentally prevent you from poirting Micronucleus to the new ATtiny series. Indeed, the first step would be to port V-USB. The main bootloader also needs adaption to the new flash controls.

Of course, the new ATtiny all support single wire debugging with UDP, which reduces the usefulness of MN somewhat.
